daemon/poller.c

```diff
@@ -537,7 +600,31 @@ void poller_timer_loop(void *d) {
 	}
 }
 
+static void sleep_ms(int ms) {
+	struct timespec deadline;
+        long next_tick;
+        clock_gettime(CLOCK_MONOTONIC, &deadline);
+
+        next_tick = (deadline.tv_sec * 1000000000L + deadline.tv_nsec) + ms * 1000000;
+        deadline.tv_sec = next_tick / 1000000000L;
+        deadline.tv_nsec = next_tick % 1000000000L;
+
+        clock_nanosleep(CLOCK_MONOTONIC, TIMER_ABSTIME, &deadline, NULL);
+}
+
 void poller_loop(void *d) {
+	struct poller_map *map = d;
+	poller_map_add(map);
+	struct poller *p = poller_map_get(map);
+
+	while (!rtpe_shutdown) {
+		int ret = poller_poll(p, 100);
+		if (ret < 0)
+			sleep_ms(20);
```

Comment:
I tried to create some kind of an thread affinity, if thread A creates another media socket, then assign itself to that socket by getting a poller from map belonging to the same thread ID.

The purpose of the sleep is to prevent excess CPU usage in _poller_poll_:
<snip>
        mutex_lock(&p->lock);

        ret = -1;
        if (!p->items || !p->items_size)
                goto out;
<snip>

if p->items_size is 0 or p->items is NULL this will bail out and be called immediately again.

If I make this configurable, is there still a chance for you to accept it?